Orange/widgets/visualize/utils/plotutils.py

```diff
@@ -0,0 +1,151 @@
+from AnyQt.QtWidgets import QScrollArea, QSizePolicy, QAction
+from AnyQt.QtCore import Qt
+from AnyQt.QtGui import QKeySequence
+
+from Orange.widgets import gui
+
+
+JITTER_SIZES = [0, 0.1, 0.5, 1, 2, 3, 4, 5, 7, 10]
+
+
+def create_scroll_area(master):
+    scroll = QScrollArea(parent=master.controlArea)
+    groupbox = gui.widgetBox(scroll, "")
+    scroll.setWidget(groupbox)
+    scroll.setWidgetResizable(True)
+    master.controlArea.layout().addWidget(scroll)
+    return groupbox
+
+
+class Jittering:
+    def __init__(self, box_plot, master, jitter_sizes=None):
+        self.jitter_sizes = jitter_sizes if isinstance(jitter_sizes, list) else JITTER_SIZES
+        self.box_plot = box_plot
+        self.master = master
+
+    def __call__(self, is_cb_jitter_cont=False):
+        gui.valueSlider(
+            widget=self.box_plot, master=self.master, value='graph.jitter_size',
+            label="Jittering: ", values=self.jitter_sizes, callback=self.master.reset_graph_data,
+            labelFormat=lambda x:
+            "None" if x == 0 else ("%.1f %%" if x < 1 else "%d %%") % x)
+        if is_cb_jitter_cont:
+            gui.checkBox(
+                widget=gui.indentedBox(widget=self.box_plot),
+                master=self.master, value="graph.jitter_continuous",
+                label="Jitter numeric values", callback=self.master.reset_graph_data)
+
+
+class Box:
+    def __init__(self, master, groupbox=None):
```

Comment:
`groupbox` has a default (`None`), but must never be `None`. Hence, it shouldn't have the default. The name is also misleading: it's not the box of the group (groupbox) but the box into which the group is put. `parent` may be a better name.